src/pages/Registration/Registration.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+
+import registerUser from '@/apis/registerUser';
+import { ERROR_CODES, STATUS_CODES } from '@/constants/constantValues';
+import { registrationSchema } from '@/helpers/validators';
+import { ServerResponse, User } from '@/types/types';
+
+import './Registration.css';
+
+export default function Registration() {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm({ resolver: yupResolver(registrationSchema) });
+  const [serverResponse, setServerResponse] = useState<ServerResponse>({ code: STATUS_CODES.OK, message: '' });
+
+  const onSubmit = handleSubmit((user: User) => {
+    registerUser(user)
+      .then((res) => {
+        if (res) {
+          setServerResponse({ code: STATUS_CODES.OK, message: 'User is created successfully, please login.' });
+          reset();
+        }
+      })
+      .catch((err) => {
+        if (err.errors[0].extensions.code === ERROR_CODES.RECORD_NOT_UNIQUE) {
+          setServerResponse({ code: STATUS_CODES.BAD_REQUEST, message: 'User with that email already exists.' });
+        }
+      });
+  });
+
+  return (
+    <div className='registration-container'>
+      <form onSubmit={onSubmit}>
+        <fieldset>
+          <h2>Registration</h2>
+
+          <div className='field'>
```

Comment:
- Move this to a separate React Field component
- Then detect common props that should be passed to each component
- Move common configuration to, for example, `REGISTRATION_FIELDS` object/array.
- Be creative :D